ts/src/connections/tunnelClientBase.ts

```diff
@@ -236,6 +250,36 @@ export class TunnelClientBase
         if (this.connectionProtocol === webSocketSubProtocolv2) {
             pfs.messageFactory = this;
             pfs.onForwardedPortConnecting((e) => this.onForwardedPortConnecting(e));
+            pfs.remoteForwardedPorts.onPortAdded((e) => this.onForwardedPortAdded(pfs, e));
+            pfs.remoteForwardedPorts.onPortUpdated((e) => this.onForwardedPortAdded(pfs, e));
+        }
+    }
+
+    private onForwardedPortAdded(pfs: PortForwardingService, e: ForwardedPortEventArgs) {
+        const port = e.port.remotePort;
+        if (typeof port !== 'number') {
+            return;
+        }
+
+        // If there are disconnected streams for the port, re-connect them now.
+        const disconnectedStreamsCount = this.disconnectedStreams.get(port)?.length ?? 0;
+        for (let i = 0; i < disconnectedStreamsCount; i++) {
+            pfs.connectToForwardedPort(port).catch((error) => {
```

Comment:
if this succeeds on step `i==0`, but then subsequently fails on step `i==1`, do we need to do anything about the successful `i==0` reconnection?

Reply:
No. Any successfully reconnected streams will be usable. For any that failed to reconnect, the client will have to retry. Note a failure does not break out of the loop -- all the operations are async anyway.